source/core/Date.cpp

```diff
@@ -113,22 +167,21 @@ namespace Vireo {
         return currentYear;
     }
 
-#if kVireoOS_windows
-    std::string abbreviateTimeZoneName(std::string input)
+    void abbreviateTimeZone(const char * input, char output[kTempCStringLength])
     {
         char prev = ' ';
-        std::string output;
-        for (int i = 0; i < input.length(); i++)
+        int j = 0;
+        for (int i = 0; i < strlen(input); i++)
         {
             if (prev == ' ' && input[i] != ' ')
             {
-                output += input[i];
+                output[j] = input[i];
+                ++j;
             }
             prev = input[i];
         }
-        return output;
+        output[j] = 0;
```

Comment:
I don't have objections to your C string refactor of this, but is this really the only way to get an abbreviated time zone on Windows? (I think Devin did this originally.)
Did you look for a different API that doesn't require use to hackishly abbreviate the string by looking for spaces?